tests/component/component.cpp

```diff
@@ -260,23 +263,23 @@ TEST(Component, removeComponentMethods)
     // No longer a component at position 1.
     EXPECT_FALSE(c->removeComponent(1));
 
-    // This will only add the one child 'c1'.
+    // This will add three children 'c1'.
     c->addComponent(c1);
     c->addComponent(c1);
     c->addComponent(c1);
     // Remove the occurrence of "child1".
     EXPECT_TRUE(c->removeComponent("child1"));
-    // Cannot remove a second occurrence of "child1".
-    EXPECT_FALSE(c->removeComponent(c1));
-    // Still have component 'c2'.
-    EXPECT_EQ(size_t(1), c->componentCount());
+    // Can remove a second occurrence of "child1".
+    EXPECT_TRUE(c->removeComponent(c1));
```

Comment:
Am I to understand that another call to `c->removeComponent(c1)` would work, but an additional one would fail? If so, I find it very confusing that after adding `c1` three times, I can remove it three times using the same reference. I could, at any point, modify `c1`, which means that when serialising the model, I will have three instances of `c1` in `c`...? Yup, I don't like it.

Reply:
I don't like it either but I think @kerimoyle  is correct in that we are thinking here about the content that is contained.  It is odd behaviour but (and this for me is important) this is not something that a person would normally do.  It doesn't make sense to add the same entity multiple times.  It just doesn't make sense to add the same units multiple times to a model.

Reply:
If it doesn't make sense to add the same entity multiple times, then why do we allow it? What is fundamentally wrong with preventing `c1` from being added multiple times to `c`?